src/u2f/u2f_app.c

```diff
@@ -29,6 +29,108 @@ static const app_t _apps[] = {
                   "\x59\x43\x79\x4e\x88\x4f\x3d\x02\x3a\x82\x29\xfd",
         .name = "Google",
     },
+    {
+        // sha256('https://bitbucket.org')
+        .app_id = "\x12\x74\x3b\x92\x12\x97\xb7\x7f\x11\x35\xe4\x1f\xde\xdd\x4a\x84\x6a\xfe\x82\xe1"
+                  "\xf3\x69\x32\xa9\x91\x2f\x3b\x0d\x8d\xfb\x7d\x0e",
+        .name = "Bitbucket",
+    },
+    {
+        // sha256('https://www.bitfinex.com')
+        .app_id = "\x30\x2f\xd5\xb4\x49\x2a\x07\xb9\xfe\xbb\x30\xe7\x32\x69\xec\xa5\x01\x20\x5c\xcf"
+                  "\xe0\xc2\x0b\xf7\xb4\x72\xfa\x2d\x31\xe2\x1e\x63",
+        .name = "Bitfinex",
+    },
+    {
+        // sha256('https://vault.bitwarden.com/app-id.json')
+        .app_id = "\xa3\x4d\x30\x9f\xfa\x28\xc1\x24\x14\xb8\xba\x6c\x07\xee\x1e\xfa\xe1\xa8\x5e\x8a"
+                  "\x04\x61\x48\x59\xa6\x7c\x04\x93\xb6\x95\x61\x90",
+        .name = "Bitwarden",
+    },
+    {
+        // sha256('https://www.dashlane.com')
+        .app_id = "\x68\x20\x19\x15\xd7\x4c\xb4\x2a\xf5\xb3\xcc\x5c\x95\xb9\x55\x3e\x3e\x3a\x83\xb4"
+                  "\xd2\xa9\x3b\x45\xfb\xad\xaa\x84\x69\xff\x8e\x6e",
+        .name = "Dashlane",
+    },
+    {
+        // sha256('https://www.dropbox.com/u2f-app-id.json')
+        .app_id = "\xc5\x0f\x8a\x7b\x70\x8e\x92\xf8\x2e\x7a\x50\xe2\xbd\xc5\x5d\x8f\xd9\x1a\x22\xfe"
+                  "\x6b\x29\xc0\xcd\xf7\x80\x55\x30\x84\x2a\xf5\x81",
+        .name = "Dropbox",
+    },
+    {
+        // sha256('https://api-9dcf9b83.duosecurity.com')
+        .app_id = "\xf3\xe2\x04\x2f\x94\x60\x7d\xa0\xa9\xc1\xf3\xb9\x5e\x0d\x2f\x2b\xb2\xe0\x69\xc5"
+                  "\xbb\x4f\xa7\x64\xaf\xfa\x64\x7d\x84\x7b\x7e\xd6",
+        .name = "Duo",
+    },
+    {
+        // sha256('facebook.com')
+        .app_id = "\x31\x19\x33\x28\xf8\xe2\x1d\xfb\x6c\x99\xf3\x22\xd2\x2d\x7b\x0b\x50\x87\x78\xe6"
+                  "\x4f\xfb\xba\x86\xe5\x22\x93\x37\x90\x31\xb8\x74",
```

Comment:
It seems facebook works only with FIDO2, and trezor also [specifies it](https://github.com/trezor/trezor-firmware/blob/65b0eae89df72eff1561d4de7221c8e34562acd4/common/defs/fido/facebook.json#L3) as webauthn.

In U2F,

> In the Web case, the FacetID MUST be the Web Origin [RFC6454] of the web page triggering the FIDO operation

(from https://fidoalliance.org/specs/fido-u2f-v1.2-ps-20170411/fido-appid-and-facets-v1.2-ps-20170411.html#determining-the-facetid-of-a-calling-application)

I would hold off adding this until https://github.com/digitalbitbox/bitbox02-firmware/pull/437/files is submitted. This won't work otherwise, I believe, and would make things confusing.

Reply:
(web origin includes the scheme so this would've been `https://facebook.com` if facebook supported it)

Reply:
U2F (BitBox02) works with Facebook. When I authenticate it shows me the sha256 provided above and url is "facebook.com".

Reply:
@x1ddos Something is weird here, can you verify this?

Reply:
Verified, the facebook app id is correct. Very good.

Reply:
I guess I was wrong and facebook's app ID isn't a URL. So, this holds:

> 1. If the AppID is not an HTTPS URL, and matches the FacetID of the caller, no additional processing is necessary and the operation may proceed.

It just so happens that `facebook.com` looks like part of a URL. Although, in this case couldn't I just fake facebok's FacetID? Since it isn't a URL, web origin is not enforced.

Reply:
They do claim to support U2F though, so I guess it's up to facebook.
https://www.facebook.com/help/401566786855239

Reply:
Could it be that webauthn is backwards compatible?

Reply:
> Could it be that webauthn is backwards compatible?

I guess so - ctrl+f "compatibility" on https://www.yubico.com/blog/what-is-fido2/ and https://fidoalliance.org/specifications/ seems to match that as well - but my concern was mostly about whether to allow non-URL facet IDs like `facebook.com` to match U2F app IDs.

We're working on fido2 support in https://github.com/digitalbitbox/bitbox02-firmware/pull/437.

Reply:
I see, Web facets must use HTTPS when it comes to U2F. 
Should I remove Facebook from the list?

Reply:
right. web facets start with the scheme and there's also `android:...` and `ios:...` for android and ios apps but facebook is none of those.

If there are currently no users who want to log in to facebook with the bitbox, I'd probably hold off adding this. Not sure. I defer to @benma. :)

Reply:
I'm using it with facebook atm, however I don't mind waiting.

I guess the right thing would be to not add it yet until FIDO2 support. If there are no downsides when it comes to security I guess it could be added.

Maybe @benma knows 😄

Reply:
Since we're in doubt, let's remove it for now. Sorry for the trouble.

Reply:
Alright, removed :)

Reply:
To clear the doubts here: no security implications really in the end but some potential troubles for existing users when using U2F on WebAuthn websites like `facebook.com`. We'll continue this in https://github.com/digitalbitbox/bitbox02-firmware/issues/474 - similar case but with `github.com`.

Reply:
(I still think it was best to remove `facebook.com` from this PR)